lib/workos/mfa.rb

```diff
@@ -0,0 +1,162 @@
+# frozen_string_literal: true
+# typed: true
+
+require 'net/http'
+require 'uri'
+
+module WorkOS
+
+  module MFA
+    class << self
+      extend T::Sig
+      include Base
+      include Client
+      
+      sig { params(id: String).returns(T::Boolean) }
+      def delete_factor(id:)
+        response = execute_request(
+            request: delete_request(
+              path: "/auth/factors/#{id}",
+              auth: true,
+            ),
+        )
+        response.is_a? Net::HTTPSuccess
+      end
+
+      sig do
+        params(
+          id: String,
+        ).returns(WorkOS::Factor)
+      end
+      def get_factor(
+          id:
+        )
+        response = execute_request(
+            request: get_request(
+              path: "/auth/factors/#{id}",
+              auth: true,
+            ),
+        )
+        WorkOS::Factor.new(response.body)
+      end
+
+      sig do
+        params(
+            type: String,
+            totp_issuer: T.nilable(String),
+            totp_user: T.nilable(String),
+            phone_number: T.nilable(String),
+        ).returns(WorkOS::Factor)
+      end
+      def enroll_factor(
+          type:,
+          totp_issuer: nil,
+          totp_user: nil,
+          phone_number: nil
+        )
+
+        if type != "sms" and type != "totp" and type != "generic_otp"
+            raise ArgumentError, "Type parameter must be either 'sms' or 'totp'"
+        end
+
+        if (type == "totp" and totp_issuer == nil) or (type == "totp" and totp_user == nil)
+            raise ArgumentError, "Incomplete arguments. Need to specify both totp_issuer and totp_user when type is totp"
+        end
+
+        if type == "sms" and phone_number == nil
+            raise ArgumentError, "Incomplete arguments. Need to specify phone_number when type is sms"
+            
+        end
+        # this condition is added as type is sms will return a 500 if content type is application/json
```

Comment:
I think you may be getting this error because the phone number is invalid. An invalid phone number will return a 500 error. And since you are calling this API with vcr/cassette and not a full mock it will give you 500 back.

Reply:
If this is what it is, we can get rid of the whole separate sms type request

Reply:
good find!  that was resolved the issue and I can would remove the if condition as well and tests still pass